olive/cli/capture_onnx.py

```diff
@@ -0,0 +1,228 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+from typing import ClassVar, Dict
+
+from olive.cli.base import (
+    BaseOliveCLICommand,
+    add_remote_options,
+    get_model_name_or_path,
+    is_remote_run,
+    update_remote_option,
+)
+from olive.common.utils import IntEnumBase, set_tempdir
+
+logger = logging.getLogger(__name__)
+
+
+class ModelBuilderAccuracyLevel(IntEnumBase):
+    fp32 = 1
+    fp16 = 2
+    bf16 = 3
+    int8 = 4
+
+
+class CaptureOnnxGraphCommand(BaseOliveCLICommand):
+    allow_unknown_args: ClassVar[bool] = True
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "capture-onnx-graph",
+            help=("Capture ONNX graph using PyTorch Exporter or Model Builder from the Huggingface model."),
+        )
+
+        # model options
+        model_group = sub_parser.add_argument_group("model options")
+        model_group.add_argument(
+            "-m",
+            "--model_name_or_path",
+            type=str,
+            required=True,
+            help=(
+                "The model checkpoint for weights initialization. If using an AzureML Registry model, provide the model"
+                " path as 'registry_name:model_name:version'."
+            ),
+        )
+        model_group.add_argument("-t", "--task", type=str, help="Task for which the model is used.")
+
+        sub_parser.add_argument("-o", "--output_path", type=str, default="onnx-model", help="Output path")
+        sub_parser.add_argument(
+            "--tempdir", default=None, type=str, help="Root directory for tempfile directories and files"
+        )
+
+        # PyTorch Exporter options
+        pte_group = sub_parser.add_argument_group("PyTorch Exporter options")
```

Comment:
I presume these options will be used by many other cli commands. Can we centralize/generalize this in one place so adding a new option to the Pass doesn't mean having to update 10 other places.